---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# fishphylo

<!-- badges: start -->
<!-- badges: end -->

The goal of fishphylo is to import sequence data from GenBank and from Fasta files and build basic phylogenetic trees.

## Installation

You can install the development version from [GitHub](https://github.com/) with:

``` {r, eval = TRUE, results='hide'}
# install.packages("devtools")
devtools::install_github("jurenoult/fishphylo")
```
# Use
## import sequences from GenBank and export a .fasta file

Load this project as a library:
```{r example}
library(fishphylo)
```

indicate taxa and the gene name
```{r}
ls_tax_gen <- cb_taxa_gene(c("Pomatoschistus lozanoi","Pomatoschistus adriaticus"),"COI")
ls_tax_gen
```

write a fasta file
```{r}
ls_fasta <- build_fasta(ls_tax_gen)
write_fasta(ls_fasta,"pomlozANDpomadri_COI_seqs.fasta")
```

## Combine sequences from two fasta files

Read a fasta file generated by write_fasta function (from GenBank), and another fasta file located in the data folder (e.g., sequence sent by Laure) 
```{r}
fas1 <- read_fasta("pomlozANDpomadri_COI_seqs.fasta")
fas2 <- read_fasta("20201104_COI_fish.fasta")
```

Show the Accession numbers and species name
```{r}
disp_access(fas1)
```
